### ISSUE TITLE
enum_variant_names triggers when used with name-aware proc macros

### ISSUE THREAD

**Issue by iitalics** [rust-lang/rust-clippy#9998]
### Summary

`enum_variant_names` is a false positive whenever it's used with proc macros such as `structopt` or `serde`, since the enum variant names are part of the UI or over-the-wire protocol being generated, not simply a matter of programmer style.

### Lint Name

enum_variant_names

### Reproducer

```rust
#[derive(StructOpt)]
enum Command {
    /// Pets a cat.
    PetCat,
    /// Pets a dog.
    PetDog,
    /// Pets a sheep.
    PetSheep,
}
/*
USAGE:
    animal-simulator <SUBCOMMAND>

FLAGS:
    -h, --help       Prints help information
    -V, --version    Prints version information

SUBCOMMANDS:
    help         Prints this message or the help of the given subcommand(s)
    pet-cat      Pets a cat
    pet-dog      Pets a dog
    pet-sheep    Pets a sheep
*/
```

One could argue about whether or not this prefix is appropriate from a UI/UX point of view, but I don't think that this is an appropriate decision for clippy to make about the program's behavior when it is intended to be enforcing code style.

### Version

```text
rustc 1.64.0 (a55dd71d5 2022-09-19)
binary: rustc
commit-hash: a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52
commit-date: 2022-09-19
host: x86_64-unknown-linux-gnu
release: 1.64.0
LLVM version: 14.0.6
```
